File: backend/app/schemas/requirements.py
from datetime import datetime
from typing import List, Optional

from pydantic import BaseModel, ConfigDict, Field


class RequirementCreate(BaseModel):
    title: str
    description: str
    focus_tags: List[str] = Field(default_factory=list)


class RequirementOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    title: str
    description: str
    focus_tags: List[str] = Field(default_factory=list)
    created_at: datetime
    updated_at: datetime

File: frontend/src/api/requirements.ts
import { api } from './client'

export interface RequirementItem {
  id: number
  title: string
  description: string
  focus_tags: string[]
  created_at: string
  updated_at: string
}

export function listRequirements() {
  return api<RequirementItem[]>('/api/requirements')
}

export function getRequirement(id: number) {
  return api<RequirementItem>(`/api/requirements/${id}`)
}
